index.js

```diff
@@ -162,6 +164,87 @@ const inventoryCommand = (message) => {
   });
 };
 
+const lookUpCommand = (message) => {
+  const originalMessage = message.content;
+  const fuzzySearchKey = originalMessage.slice(8);
+
+  axios
+    .get(
+      "https://db.ygoprodeck.com/api/v7/cardinfo.php?&fname=" + fuzzySearchKey
+    )
+    .then(function (cardSearchResults) {
+      console.log(
+        "https://db.ygoprodeck.com/api/v7/cardinfo.php?&fname=" + fuzzySearchKey
+      );
+
+      let searchedCardName;
+      let closestDataIndex = -2;
+      let secondClosestDataIndex;
+      cardSearchResults.data.data.forEach((element, index) => {
+        if (element.name.toLowerCase() === fuzzySearchKey.toLowerCase()) {
+          closestDataIndex = index;
+        } else if (
+          element.name.toLowerCase().startsWith(fuzzySearchKey.toLowerCase())
+        ) {
+          secondClosestDataIndex = index;
+        } else {
+          console.log("This didn't work");
+        }
+      });
+
+      if (closestDataIndex > -1) {
+        const attachment = new Discord.MessageAttachment(
+          cardSearchResults.data.data[closestDataIndex].card_images[0].image_url
+        );
+        searchedCardName = cardSearchResults.data.data[closestDataIndex].name;
+        message.reply(
+          "The top result for your search is: " +
+            cardSearchResults.data.data[closestDataIndex].name
+        );
+        message.channel.send(attachment);
+      } else if (closestDataIndex < -1) {
+        const attachment = new Discord.MessageAttachment(
+          cardSearchResults.data.data[
+            secondClosestDataIndex
+          ].card_images[0].image_url
+        );
+
+        message.reply(
+          "The closest result for your search is: " +
+            cardSearchResults.data.data[secondClosestDataIndex].name
+        );
+
+        searchedCardName =
+          cardSearchResults.data.data[secondClosestDataIndex].name;
+        message.channel.send(attachment);
+      }
+
+      const suggestedSearches = cardSearchResults.data.data
+        .filter((item) => item.name != searchedCardName)
+        .map((item, index) => {
+          if (index < 6) {
+            return "•  " + item.name;
+          } else {
+            return;
```

Comment:
`.map()` takes an array of size X and should return an array of size X.
`.filter()` takes an array of size X and should return an array of size ≤ X.

This is sort of technically working right now, but doing the limiting of 6 items would fit the model better if it were done in the `.filter` instead of the `.map`. Any time you're reducing the size of the array, that's a job for `.filter(`

Reply:
Actually even better than adding this in `.filter(` is using `.slice(`, which is made specifically for this. You can chain it just like `.filter` and `.map`, probably like `.filter().slice().map()`